flight/PiOS/STM32F4xx/pios_dacbeep.c

```diff
@@ -0,0 +1,211 @@
+/**
+ ******************************************************************************
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+  * @addtogroup   PIOS_DACBEEP DAC Beep Code Functions
+ * @brief PIOS interface for DAC Beep implementation
+ * @{
+ *
+ * @file       pios_dacbeep.c
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2015-2016
+ * @brief      Generates audio beeps of desired duration and frequency
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+/* Project Includes */
+#include "pios.h"
+#if defined(PIOS_INCLUDE_DAC_BEEPS)
+
+#if defined(PIOS_INCLUDE_FREERTOS)
+#include "FreeRTOS.h"
+#endif /* defined(PIOS_INCLUDE_FREERTOS) */
+
+#include "pios_dac_common.h"
+
+enum pios_dacbeep_dev_magic {
+	PIOS_DACBEEP_DEV_MAGIC = 0x3A53834A,
+};
+
+struct pios_dacbeep_dev {
+	enum pios_dacbeep_dev_magic     magic;
+	int32_t cycles_remaining;
+};
+
+// Lookup tables for symbols
+// from ML designed to generate 0.4V p-p centered at 0.5V
+//   sprintf('%d,', round((sin(linspace(0,2*pi,130))*(2^12/3.3*0.2)+(2^12/3.3*0.5))))
```

Comment:
Yeah - you should see the blocks of ML code embeded in the FSK_DAC code to come :)